src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -223,7 +222,7 @@ void CalorimeterHitReco::AlgorithmProcess() {
         const decltype(edm4eic::CalorimeterHitData::position) position(gpos.x() / m_lUnit, gpos.y() / m_lUnit,
                                                                     gpos.z() / m_lUnit);
         const decltype(edm4eic::CalorimeterHitData::dimension) dimension(cdim[0] / m_lUnit, cdim[1] / m_lUnit,
```

Comment:
```suggestion
        // `dimension` is 3D, but `cdim` can be 2D only
        const decltype(edm4eic::CalorimeterHitData::dimension) dimension(cdim[0] / m_lUnit, cdim[1] / m_lUnit,
```

Reply:
I assume that's not true for the `else` branch of the `if` above.